layer/executables/entrypoint/dataset.py

```diff
@@ -74,6 +74,17 @@ def _run(
             result = dataset_definition.func(
                 *dataset_definition.args, **dataset_definition.kwargs
             )
+
+            # in case client does not return the dataset
+            if result is None:
+                tracker.mark_dataset_built(dataset_definition.asset_name)
```

Comment:
Can we also add a `save_dataset` function similar to `save_model` to allow users save datasets on their own?